gossip/minimal/goshawk.go

```diff
@@ -173,30 +156,28 @@ func (hawk *Goshawk) Fly(ctx context.Context) {
 			glog.Infof("finished Checker(%s)", origin.Name)
 		}(origin)
 	}
+
 	wg.Wait()
 }
 
 // Scanner runs a continuous scan of the destination log.
 func (hawk *Goshawk) Scanner(ctx context.Context) {
-	ticker := time.NewTicker(hawk.dest.MinInterval)
-	for {
-		glog.V(1).Infof("Scanner(%s): run scan from %d to current size", hawk.dest.Name, hawk.scanOpts.StartIndex)
-		s := scanner.NewScanner(hawk.dest.Log, hawk.scanOpts)
-		size, err := s.ScanLog(ctx, hawk.foundCert, foundPrecert)
-		if err != nil {
-			glog.Errorf("Scanner(%s) terminated: %v", hawk.dest.Name, err)
-			return
-		}
-		hawk.scanOpts.StartIndex = size
-
-		glog.V(2).Infof("Scanner(%s): wait for a %s tick", hawk.dest.Name, hawk.dest.MinInterval)
-		select {
-		case <-ctx.Done():
-			glog.Infof("Scanner(%s): termination requested", hawk.dest.Name)
-			return
-		case <-ticker.C:
+	fetcher := scanner.NewFetcher(hawk.dest.Log, &hawk.fetchOpts)
+	fetcher.Run(ctx, func(batch scanner.EntryBatch) {
```

Comment:
`Run` returns an error. Maybe return this error from `Scanner()`?

Reply:
Good idea - if the fetcher fails (for something other than context cancellation) the other goroutines will carry on regardless.

Reply:
Leaving this for a follow-up PR?

Reply:
Ah, didn't notice you fixed that. Ignore my comment.